--- main.cpp
+++ main.cpp
@@ -1,18 +1,17 @@
 #include <iostream>
+#include <vector>
 
 size_t fib(size_t n)
 {
-    size_t first = 1;
-    size_t second = 1;
+    assert(n > 0);
+    std::vector<size_t> res = {1, 1};
     for (size_t i = 2; i < n; ++i) {
-        size_t next = first + second;
-        first = second;
-        second = next;
+        res.push_back(res[i - 1] + res[i - 2]);
     }
-    return second;
+    return res[n - 1];
 }
 
 int main()
 {
     std::cout << fib(6) << std::endl;
 }
